core/lib/src/catcher/catcher.rs

```diff
@@ -132,10 +131,11 @@ pub struct Catcher {
     pub(crate) location: Option<(&'static str, u32, u32)>,
 }
 
-// The rank is computed as -(number of nonempty segments in base) => catchers
+// The rank is computed as -(number of nonempty segments in base) *2 => catchers
 // with more nonempty segments have lower ranks => higher precedence.
+// Doubled to provide space between for typed catchers.
 fn rank(base: Path<'_>) -> isize {
-    -(base.segments().filter(|s| !s.is_empty()).count() as isize)
+    -(base.segments().filter(|s| !s.is_empty()).count() as isize) * 2
 }
```

Comment:
This logic has been fully removed. The preference for typed catchers is now implemented in lifecycle.rs, since it also needs to check the error's `.source()` method, it now always checks for a matching type before a catcher with any type.